src/mesh.c

```
@@ -6,381 +6,325 @@
+            }
+            break;
         }
-        break;
-    }
 
-    /* Perform periodic tasks */
-    mesh_periodic_tasks();
+        case PACKET_TYPE_GROUP_JOIN: {
+            /* Payload is already decrypted in-place by mesh_rx_handler using the pairwise session key. */
+            uint16_t sender_id = packet->header.source;
+            if (packet->payload_len < sizeof(group_join_payload_t)) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)